Source/WebCore/platform/image-decoders/bmp/BMPImageReader.cpp

```diff
@@ -621,9 +629,9 @@ bool BMPImageReader::processRLEData()
                 }
                 if ((colorIndexes[0] >= m_infoHeader.biClrUsed) || (colorIndexes[1] >= m_infoHeader.biClrUsed))
                     return m_parent->setFailed();
-                for (int which = 0; m_coord.x() < endX; ) {
+                for (unsigned which = 0; m_coord.x() < endX;) {
                     setI(colorIndexes[which]);
-                    which = !which;
+                    which ^= 1;
```

Comment:
Did you check disassembled code? How does it look different?
https://godbolt.org/z/MYhYW5sva

Reply:
which is an int, not a bool so yeah it will be different.

Reply:
What compiler are you using? Could you let me see your disassemble code of my program? https://godbolt.org/z/MYhYW5sva